src/packet/route/route.rs

```diff
@@ -283,6 +289,58 @@ impl<'a> ToPayload for RtAttrPacket<'a> {
     }
 }
 
+impl<'a> ToPayload for &'a [&'a Nested<'a>] {
+    fn payload_add(&self, payload: &mut [u8]) {
+        self.iter().fold(0, |pos, pkg| {
+            pkg.payload_add(&mut payload[pos..]);
+            pos + pkg.payload_size()
+        });
+    }
+
+    fn payload_size(&self) -> usize {
+        self.iter().map(|p| p.payload_size()).sum()
+    }
+}
+
+impl<'a> ToPayload for &'a [u8] {
+    fn payload_add(&self, payload: &mut [u8]) {
+        payload[..self.len()].copy_from_slice(self)
+    }
+
+    fn payload_size(&self) -> usize {
+        self.len() + 1
+    }
+}
+
+impl<'a> ToPayload for Nested<'a> {
+    fn payload_add(&self, payload: &mut [u8]) {
+        match *self {
+            Nested::RtAttr(ref r) => payload[..r.packet_size()].copy_from_slice(&r.packet()),
+            Nested::InfoAttr(ref i) => payload[..i.packet_size()].copy_from_slice(&i.packet()),
+        }
+    }
+
+    fn payload_size(&self) -> usize {
+        match *self {
+            Nested::RtAttr(ref r) => util::align(r.packet_size()),
+            Nested::InfoAttr(ref i) => util::align(i.packet_size()),
+        }
+    }
+}
+
+impl<'a> ToPayload for &'a [&'a RtAttrPacket<'a>] {
```

Comment:
As we already implement impl<'a> ToPayload for &'a [&'a ToPayload, this implementation should already exist implicit.